### ISSUE TITLE
Need an easy way to sign a message

### ISSUE THREAD

**Issue by bota87** [joinmarket-webui/jam#633]
**Is your feature request related to a problem? Please describe.**

Some services require you to sign a message with the address you want to use to receive

**Describe the solution you'd like**

A tool that ask for the address, the message to sign and return the signature, like sparrow/electrum wallets

**Comment by theborakompanioni** [comment]
Hey @bota87! Thanks for the feature request.
It would be nice to have this ability in Jam, however, such functionality is currently not exposed via the JM API.
Would it be sufficient to use the CLI? I think `wallet-tool.py` should expose the command `signmessage` with should be what you are looking for.

**Comment by bota87** [comment]
Yeah I've solved with wallet-tool but it's a little bit tricky, especially having to set the hd wallet path instead of the address.
`docker exec -it jam wallet-tool.py -H "m/84'/0'/2'/0/2" /root/.joinmarket/wallets/jam_wallet.jmdat signmessage "the message to sign"`

**Comment by theborakompanioni** [comment]
> Yeah I've solved with wallet-tool but it's a little bit tricky

:+1:

If this feature will find its way into the JM API, Jam will happily adopt it!

**Comment by bota87** [comment]
Ok I've opened an issue on JM repo, let's see how it goes :crossed_fingers:

**Comment by theborakompanioni** [comment]
Ready for development. See https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1536 (not merged yet).

**Comment by kristapsk** [comment]
> Ready for development. See [JoinMarket-Org/joinmarket-clientserver#1536](https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1536) (not merged yet).

Waiting for feedback from you or somebody else that it works well with Jam, then will merge.

**Comment by theborakompanioni** [comment]
ping @editwentyone

**Comment by editwentyone** [comment]
new update after call:
https://www.figma.com/file/kfejZJFlwBywvLEnPEmJo1/JoinMarket-UI?type=design&node-id=6600-17405&mode=design&t=5062kYXLtdJFrL4Y-11

<img width="430" alt="image" src="https://github.com/joinmarket-webui/jam/assets/96262473/22427ab8-8e6e-4e35-973c-a4380cd860bb">

**Comment by kunal-595** [comment]
Hi @theborakompanioni,
I’d like to work on this issue as my contribution to Jam.
Once the JM API merge is ready, I plan to implement a simple UI to sign messages with a Bitcoin address and return the signature.
Could you please assign this issue to me?
Thanks!

**Comment by theborakompanioni** [comment]
> Hi [@theborakompanioni](https://github.com/theborakompanioni), I’d like to work on this issue as my contribution to Jam. Once the JM API merge is ready, I plan to implement a simple UI to sign messages with a Bitcoin address and return the signature. Could you please assign this issue to me? Thanks!

Since the upstream changes are not merged yet, this is still blocked.
Please contact me on matrix and we will find something to work on in the new v2 branch 🙏

**Comment by kristapsk** [comment]
> Since the upstream changes are not merged yet

I could merge them as is if it's useful and works for Jam in current form.

**Comment by theborakompanioni** [comment]
> > Since the upstream changes are not merged yet
> 
> I could merge them as is if it's useful and works for Jam in current form.

It seems the feature is not in high demand, but the use case is valid. If you decide to merge, we will make an effort to support it.

**Comment by bobben416** [comment]
I think this feature is very helpful - in Europe everyone who ever wants to send funds to an exchange or even prove the tax status of funds needs to be able to sign adresses. It would be very helpful to be able to sign joinmarket adresses - bond adresses included.

**Comment by kristapsk** [comment]
> in Europe everyone who ever wants to send funds to an exchange (...) needs to be able to sign adresses

Offtopic - doesn't affect non-custodial / P2P exchanges.

**Comment by kristapsk** [comment]
> > > Since the upstream changes are not merged yet
> > 
> > 
> > I could merge them as is if it's useful and works for Jam in current form.
> 
> It seems the feature is not in high demand, but the use case is valid. If you decide to merge, we will make an effort to support it.

https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1536 has been merged into master.

**Comment by TomLucasakaTGeek** [comment]
is this issue resolved @kristapsk @theborakompanioni ??

**Comment by theborakompanioni** [comment]
> is this issue resolved [@kristapsk](https://github.com/kristapsk) [@theborakompanioni](https://github.com/theborakompanioni) ??

No, not yet implemented.
